src/consumeroffset/kafka_offset_collect.go

```diff
@@ -221,59 +221,83 @@ func createFetchRequest(topicPartitions TopicPartitions, client connection.Clien
 	return request
 }
 
-// populateOffsetStructs takes a map of offsets and high water marks and
-// populates an array of partitionOffsets which can then be marshalled into metric
-// sets
-func populateOffsetStructs(offsets, hwms groupOffsets) []*partitionOffsets {
-
-	var poffsets []*partitionOffsets
-	for topic, partitions := range hwms {
-		for partition, hwm := range partitions {
-			offsetPointer := func() *int64 {
-				topicOffsets, ok := offsets[topic]
-				if !ok || len(topicOffsets) == 0 {
-					log.Error("Offset not collected for topic %s", topic, partition)
-					return nil
+func collectOffsetsForConsumerGroup(client connection.Client, clusterAdmin sarama.ClusterAdmin, consumerGroup string, members map[string]*sarama.GroupMemberDescription, kafkaIntegration *integration.Integration, wg *sync.WaitGroup) {
```

Comment:
currently it implements only option "compute the totals"

Advantages: Now stalled consumer are included into the total
Disadvantages: Since the metric is now included some user could be surprised by the change in the totals (possible breaking change)